cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/runnable/RightsizeRunnable.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.handler.async.runnable;
+
+import com.linkedin.kafka.cruisecontrol.KafkaCruiseControl;
+import com.linkedin.kafka.cruisecontrol.analyzer.ProvisionRecommendation;
+import com.linkedin.kafka.cruisecontrol.analyzer.ProvisionStatus;
+import com.linkedin.kafka.cruisecontrol.common.Utils;
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import com.linkedin.kafka.cruisecontrol.config.constants.AnomalyDetectorConfig;
+import com.linkedin.kafka.cruisecontrol.detector.Provisioner;
+import com.linkedin.kafka.cruisecontrol.detector.ProvisionerState;
+import com.linkedin.kafka.cruisecontrol.detector.RightsizeOptions;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.RightsizeParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.response.RightsizeResult;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static com.linkedin.kafka.cruisecontrol.detector.AnomalyDetectorUtils.KAFKA_CRUISE_CONTROL_OBJECT_CONFIG;
+
+
+/**
+ * The async runnable for running a provision rightsize request
+ */
+public class RightsizeRunnable extends OperationRunnable {
+  private static final String RECOMMENDER_UP = "Recommender-Under-Provisioned";
+  protected final RightsizeParameters _parameters;
+
+  public RightsizeRunnable(KafkaCruiseControl kafkaCruiseControl, OperationFuture future, RightsizeParameters parameters) {
+    super(kafkaCruiseControl, future);
+    _parameters = parameters;
+  }
+
+  @Override
+  protected RightsizeResult getResult() throws Exception {
```

Comment:
Does this method throw an `Exception`? If not can we drop `throws Exception`?

Reply:
Removing this class to set the Rightsize endpoint as a synchronous request. The content of this `RightsizeRunnable#getResult` method will be moved the the `RightsizeResult#handle` method